tools/pyi/gen_pyi.py

```diff
@@ -300,22 +303,27 @@ def generate_type_hints(fname, decls, is_tensor=False):
         numargs = len(decl['arguments'])
         vararg_pos = int(is_tensor)
         have_vararg_version = (numargs > vararg_pos and
-                               decl['arguments'][vararg_pos]['dynamic_type'] in {'IntArrayRef', 'TensorList'} and
+                               decl['arguments'][vararg_pos]['dynamic_type'] in {'IntArrayRef', 'TensorList', 'DimnameList'} and
                                (numargs == vararg_pos + 1 or python_args[vararg_pos + 1] == '*') and
                                (not is_tensor or decl['arguments'][0]['name'] == 'self'))
 
         type_hints.append(type_hint)
 
         if have_vararg_version:
             # Two things come into play here: PyTorch has the "magic" that if the first and only positional argument
-            # is an IntArrayRef or TensorList, it will be used as a vararg variant.
+            # is an IntArrayRef, TensorList, or DimnameList, it will be used as a vararg variant.
```

Comment:
I'm not sure what is going on here. If I am understanding this correctly, this is only used to generate type annotations for `tensor.align_to`. If that is the case, could we manually bind it instead?

Reply:
I started off trying to handle the generic case, and then realized that almost every method required a manual binding.  If more variable argument methods are added to NamedTensor in the future this code should result in automatic bindings, so there is some utility in leaving it in.  Let me know if you think that's unlikely to happen and I'm happy to strip out the changes to the generated and add the manual case for `align_to`.

Reply:
I don't see a large number of variable argument methods being added in the near future. The manual case makes it more obvious what is going on and is not too much effort to write for each new op.